plugin/trino-clickhouse/src/main/java/io/trino/plugin/clickhouse/ClickHouseClient.java

```diff
@@ -241,6 +243,11 @@ public ClickHouseClient(
         JdbcTypeHandle bigintTypeHandle = new JdbcTypeHandle(Types.BIGINT, Optional.of("bigint"), Optional.empty(), Optional.empty(), Optional.empty(), Optional.empty());
         this.connectorExpressionRewriter = JdbcConnectorExpressionRewriterBuilder.newBuilder()
                 .addStandardRules(this::quoted)
+                .map("$equal(left: varchar, right: varchar)").to("left = right")
+                .add(new RewriteIn(type -> type instanceof VarcharType))
```

Comment:
deserves a comment on why it's limited to varchars, not obvious to me

Reply:
To limit the scope of this PR we are focusing on Varchars only. 
Eventually all types have to be supported.
Handling all types within the same PR blows up the PR and makes it not mergable. E.g. https://github.com/trinodb/trino/pull/22921